### PR TITLE
fix(macro): Fix the apply of macros for editor2 (SDESK-4002)

### PR DESCRIPTION
**Pull request by ioanpocol**

### CODE REVIEW

File: scripts/apps/authoring/macros/macros.js [hunk 1]
```diff
@@ -191,6 +191,7 @@ function MacrosController($scope, macros, desks, autosave, $rootScope, storage,
             }
 
             if (isEditor3 || res.diff == null) {
+                $rootScope.macro_dirty = true;
```
Comment (ioanpocol) [inline scripts/apps/authoring/macros/macros.js]: @petrjasek 
Any idea how to set dirty for the current item?

Comment (petrjasek) [reply scripts/apps/authoring/macros/macros.js]: what about `$scope.autosave()` ?

Comment (ioanpocol) [reply scripts/apps/authoring/macros/macros.js]: Nemanja proposed first a more generic solution to use broadcast.
Please review it.

Comment (petrjasek) [reply scripts/apps/authoring/macros/macros.js]: imo using autosave like all other components is better

Comment (ioanpocol) [reply scripts/apps/authoring/macros/macros.js]: OK, it is better autosave then. Please review again.

File: scripts/apps/authoring/authoring/directives/AuthoringTopbarDirective.ts [hunk 2]
```diff
@@ -26,6 +26,7 @@ export function AuthoringTopbarDirective(TranslationService, privileges, authori
             scope.saveTopbar = function() {
                 scope.saveDisabled = true;
                 return scope.save(scope.item)
+                    .then((item) => angular.extend(scope.item, item))
```
Comment (petrjasek) [inline scripts/apps/authoring/authoring/directives/AuthoringTopbarDirective.ts]: why is this needed here?

Comment (ioanpocol) [reply scripts/apps/authoring/authoring/directives/AuthoringTopbarDirective.ts]: to be sure that after save the item has the same value as origItem

Comment (petrjasek) [reply scripts/apps/authoring/authoring/directives/AuthoringTopbarDirective.ts]: shouldn't it be part of that `scope.save` method then?

Comment (ioanpocol) [reply scripts/apps/authoring/authoring/directives/AuthoringTopbarDirective.ts]: OK, moved on save
